File: teamAPI/src/main/java/com/ironhack/pokemon/model/Team.java
package com.ironhack.pokemon.model;

import com.ironhack.pokemon.controller.dtos.*;

import javax.persistence.*;
import java.util.*;

@Entity
public class Team {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long teamId;
    @OneToOne
    @JoinColumn(name = "trainer_id")
    private Trainer trainer;
    @ManyToMany
    @JoinTable(
            name = "pokemon_in_team",
            joinColumns = { @JoinColumn(name = "team_id") },
            inverseJoinColumns = { @JoinColumn(name = "pokemon_id") }
    )
    private List<Pokemon> pokemonList;

//    Constructor
    public Team() {
    }

    public Team(Trainer trainer, List<Pokemon> pokemonList) {
        this.trainer = trainer;
        this.pokemonList = pokemonList;
    }

    public Team(TeamDTO teamDTO){
        this(new Trainer(teamDTO.getTrainer()), teamDTO.getPokemonList());
    }

    //    Getters and Setters
    public Long getTeamId() {
        return teamId;
    }

    public void setTeamId(Long teamId) {
        this.teamId = teamId;
    }

    public Trainer getTrainer() {
        return trainer;
    }

    public void setTrainer(Trainer trainer) {
        this.trainer = trainer;
    }

    public List<Pokemon> getPokemonList() {
        return pokemonList;
    }

    public void setPokemonList(List<Pokemon> pokemonList) {
        this.pokemonList = pokemonList;
    }
}

File: pokemon/src/app/models/team.ts
import { Trainer } from './trainer';



export class Team {
    constructor(
        private _id: number,
        private _trainer: Trainer,
        private _pokemonList: {id: number, pokedexId:number}[],
    ){}

    get id(): number {
        return this._id;
    }
    get trainer(): Trainer {
        return this._trainer;
    }
    get pokemonList(): {id: number, pokedexId:number}[] {
        return this._pokemonList;
    }

    set id(id: number) {
        this._id = id;
    }
    set trainer(trainer: Trainer) {
        this._trainer = trainer;
    }
    set pokemonList(pokemonList: {id: number, pokedexId:number}[]) {
        this._pokemonList = pokemonList;
    }
}





